### PR TITLE
fix: Delete obsolete properties from phishingController and networkControl…

### PR DESCRIPTION
**Pull request by danjm**
## **Description**

Expands on https://github.com/MetaMask/metamask-extension/pull/26383 to delete more obsolete state from the Network and Phishing controllers, to eliminate other sentry errors.

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26396?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: app/scripts/migrations/120.4.test.ts [hunk 1]
```diff
@@ -95,4 +95,138 @@ describe('migration #120.4', () => {
       });
     });
   });
+
+  describe('PhishingController', () => {
+    it('does nothing if CurrencyController state is not set', async () => {
```
Comment (Gudahtt) [inline app/scripts/migrations/120.4.test.ts:100]:
```suggestion
    it('does nothing if PhishingController state is not set', async () => {
```

File: app/scripts/migrations/120.4.test.ts [hunk 2]
```diff
@@ -95,4 +95,138 @@ describe('migration #120.4', () => {
...
+      });
+
+      expect(transformedState.data).toEqual({
+        PhishingController: { phishingLists: 'test' },
+      });
+    });
+  });
+
+  describe('NetworkController', () => {
+    it('does nothing if CurrencyController state is not set', async () => {
```
Comment (Gudahtt) [inline app/scripts/migrations/120.4.test.ts:167]:
```suggestion
    it('does nothing if NetworkController state is not set', async () => {
```

**Review by Gudahtt — CHANGES_REQUESTED** [review]
Mostly looks good! Just two copy+paste test description errors

File: app/scripts/migrations/120.4.test.ts [hunk 3]
```diff
@@ -95,4 +95,138 @@ describe('migration #120.4', () => {
...
+          `Migration ${version}: Invalid NetworkController state of type 'string'`,
+        ),
+      );
+    });
+
+    it('deletes obsolete properties from the NetworkController state', async () => {
+      const oldState = {
+        NetworkController: {
+          phishing: 'test',
+          lastFetched: 'test',
```
Comment (Gudahtt) [inline app/scripts/migrations/120.4.test.ts:202]: This fixes the unit test failure
```suggestion
          network: 'test',
```

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by darkwing — APPROVED** [review]

**Review by jiexi — APPROVED** [review]

**Pull request merged**